### PR TITLE
Update Versioneer to 0.29

### PR DESCRIPTION
**Pull request by penguinpee**
Required for Python 3.12.
SafeConfigParser has been removed in Python 3.12.
(has been deprecated since Python 3.2)

### CODE REVIEW

**Comment by AurelienJaquier** [comment]
Thank you @penguinpee  for your help on improving efel. Could you also change the style in `setup.cfg` from `pep440-minor` to `pep440`? It looks like `pep440-minor` is deprecated in latest versioneer.

**Comment by penguinpee** [comment]
Style has been updated.

File: efel/_version.py [hunk 1]
```diff
@@ -43,7 +53,7 @@ def get_config():
     cfg.style = "pep440-minor"
```
Comment (AurelienJaquier) [inline efel/_version.py:53]: Could you also change here to `"pep440"` please?

**Review by AurelienJaquier — COMMENTED** [review]
Also some tests are failing because of linting. Could you take care of it?

File: efel/_version.py (hunk 1 above)
Comment (penguinpee) [reply efel/_version.py:53]: I believe that file is generated by Versioneer based on the config. I'll run that again and probably squash the two commits into one.

**Comment by penguinpee** [comment]
> Also some tests are failing because of linting. Could you take care of it?

Hmm. It complains about three lines in `efel/_version.py` being too long. Since this file is generated I'd rather not hand edit it. Can it be exempted from triggering `E501 line too long`?

**Comment by AurelienJaquier** [comment]
This looks great! Thank you very much @penguinpee I'll merge this

**Pull request merged**